vault/expiration_test.go

```diff
@@ -2839,6 +2839,9 @@ func TestExpiration_MarkIrrevocable(t *testing.T) {
 	if _, ok := exp.irrevocable.Load(leaseID); !ok {
 		t.Fatalf("irrevocable lease not included in irrevocable map")
 	}
+	if exp.irrevocableLeaseCount != 1 {
```

Comment:
I just noticed this in previous code (line 2831) - we might want to hold pending lock for some of these updates for things that are supposed to be called/accessed locked. I'm not sure why the race tests didn't go off on this earlier?

Reply:
hmm i think maybe not ? so in this test, one lease was registered then marked as irrevocable on L2831, so I don't _think_ anything else will be trying to concurrently access this value, so it should be safe to access in this section without the lock. does that make sense ?

Reply:
i think some good advice i've heard is if you have to justify why something doesn't need concurrent protection in a specific instance, it's best and most clear to just lock it

Reply:
hah that does sound like a very good rule of thumb, gotcha, i'll go ahead and lock up the `irrevocableLeaseCount` accesses in the test!

Reply:
2558e12

Reply:
cool. while you're in here, could you also lock around the stuff that was on line 2831 (if i remember right i must have forgotten to do that before calling a function that should be called under lock)

feel free to do this anywhere else you see that might be relevant!

Reply:
ah gotcha, can do!

Reply:
fixed this and a couple other tests here e6eda3d